File: android/measure/src/test/java/sh/measure/android/fakes/FakeSpanProcessor.kt
package sh.measure.android.fakes

import sh.measure.android.tracing.InternalSpan
import sh.measure.android.tracing.Span
import sh.measure.android.tracing.SpanProcessor

internal class FakeSpanProcessor : SpanProcessor {
    val startedSpans = mutableListOf<InternalSpan>()
    val endingSpans = mutableListOf<Span>()
    val endedSpans = mutableListOf<Span>()

    override fun onStart(span: InternalSpan) {
        startedSpans += span
    }

    override fun onEnding(span: InternalSpan) {
        endingSpans += span as Span
    }

    override fun onEnded(span: InternalSpan) {
        endedSpans += span as Span
    }

    override fun onConfigLoaded() {
        // no-op
    }
}

File: flutter/packages/measure_flutter/test/utils/fake_span_processor.dart
import 'package:measure_flutter/src/tracing/internal_span.dart';
import 'package:measure_flutter/src/tracing/span_processor.dart';

class FakeSpanProcessor implements SpanProcessor {
  final List<InternalSpan> startedSpans = [];
  final List<InternalSpan> endingSpans = [];
  final List<InternalSpan> endedSpans = [];

  @override
  Future<void> onStart(InternalSpan span) async {
    startedSpans.add(span);
  }

  @override
  Future<void> onEnding(InternalSpan span) async {
    endingSpans.add(span);
  }

  @override
  Future<void> onEnded(InternalSpan span) async {
    endedSpans.add(span);
  }

  int get startedSpanCount => startedSpans.length;

  int get endingSpanCount => endingSpans.length;

  int get endedSpanCount => endedSpans.length;
}
